pyodide-build/pyodide_build/install_xbuildenv.py

```diff
@@ -0,0 +1,44 @@
+import argparse
+import shutil
+import subprocess
+from pathlib import Path
+from sys import version_info
+
+from .common import get_make_flag, get_pyodide_root
+
+
+def make_parser(parser: argparse.ArgumentParser):
+    parser.description = (
+        "Install xbuild env.\n\n"
+        "Note: this is a private endpoint that should not be used "
+        "outside of the Pyodide Makefile."
+    )
+    return parser
+
+
+def main(args):
+    pyodide_root = get_pyodide_root()
+    host_site_packages = Path(get_make_flag("HOSTSITEPACKAGES"))
+    xbuildenv_path = pyodide_root / "xbuildenv"
```

Comment:
Maybe take this as CLI argument, for the case where we download it from somewhere.

Reply:
I had in mind to leave this part incomplete in this PR and fix it up in a subsequent PR. My thought was that by default could calculate the URL from the current commit hash and get it from s3.